File: backends/java/zappy-application/src/main/java/com/zappymart/application/ports/CartRepository.java
package com.zappymart.application.ports;

import com.zappymart.domain.cart.Cart;

import java.util.Optional;

public interface CartRepository {

    Optional<Cart> byId(String cartId);

    Optional<Cart> ofCustomer(String customerId);

    Cart save(Cart cart);

    void delete(String cartId);
}

File: backends/kotlin/zappy-application/src/main/kotlin/nl/zappymart/application/ports/CartRepository.kt
package nl.zappymart.application.ports

import nl.zappymart.domain.cart.Cart

interface CartRepository {

    fun findById(cartId: String): Cart?

    fun findByCustomerId(customerId: String): Cart?

    fun save(cart: Cart): Cart

    fun delete(cartId: String)
}
